include/openenclave/internal/switchless.h

```diff
@@ -10,22 +10,29 @@
 
 typedef uint64_t oe_thread_t;
 
+typedef struct _switchless_settings
```

Comment:
This cannot be an internal struct if you expect that developers will pass this into the public `oe_create_enclave` call directly. 

Also missed this earlier: please make sure your typedefs are correctly suffixed by `_t`, so `oe_switchless_settings_t`, `oe_switchless_call_manager_t`

Reply:
More generally, because we expect the `oe_create_enclave()` parameter `config` to be able to take a range of composable configuration structs, please make sure that the structs are designed as TLV (type-length-value) so that they can be walked by the `oe_create_enclave()` implementation, for example:

```c
typedef struct _oe_switchless_settings
{
    uint32_t magic;
    size_t size;
    size_t num_host_workers;
} oe_switchless_settings_t;
```